### ISSUE TITLE
Are specialized epilogues used in the compiler?

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/omr#1730]
Both P and Z code generators have support for "specialized epilogues", which appear to be an attempt to selectively restore preserved registers.  This feature is currently guarded with a compile option (`enableSpecializedEpilogues`).

Because I can see the compiler code long before it was contributed to Eclipse OMR, I can tell it hasn't been enabled on any known project for at least 5 years.  It's arguably experimental code.  Any objections to removing the code from the Z and P code generators to improve clarity?  If not, what's the justification for keeping it?  

@gita-omr @fjeremic @ymanton

**Comment by ymanton** [comment]
No objections from me, the "specialization" that was done on Power under that option was very minor anyway.

**Comment by fjeremic** [comment]
No objection from me either. Slightly related but is shrink wrapping even functionally correct at this point? IIRC the prototype never really showed performance benefit. We should probably be looking at deprecating it as well.

**Comment by gita-omr** [comment]
I agree, it can be removed.

**Comment by 0xdaryl** [comment]
Thanks for the input.  I will open a separate issue to remove and close this question.

We had a debate around shrink wrapping when we were making the initial compiler source contribution.  For everyone's edification, "shrink wrapping" is an optimization where register preservation required by a method's linkage is pushed out of the method prologues down onto the execution paths where they are truly required.  The idea is to move unnecessary register preserves out of method prologues/epilogues if the preservation is only required on cold paths in a method.

While shrink wrapping never really showed much benefit on Java workloads on any architecture, we decided to contribute the code to OMR anyway because it is a useful example of how to do a dataflow analysis over TR instructions.  We did not have a means of contributing this outside of the working code base (e.g., in an `examples` directory) and the context was important to understanding it.

However, it is a bit of a shame that this code remains tangled up with the working code base because of the seemingly needless dependencies it brings with it.  I'm open to ideas for how this code can be preserved and showcased as an example while being disconnected from the working code base.
